components/net/subresource_integrity.rs

```diff
@@ -0,0 +1,164 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+use net_traits::response::{Response, ResponseBody, ResponseType};
+use openssl::crypto::hash::{hash, Type as MessageDigest};
+use rustc_serialize::base64::{STANDARD, ToBase64};
+use std::collections::HashMap;
+use std::sync::MutexGuard;
+lazy_static! {
+    //Key is supported algorithm and value is priority
+    static ref SUPPORTED_ALGORITHM: HashMap<&'static str, usize> = {
```

Comment:
We can do this instead:
```rust
const SUPPORTED_ALGORITHM: &'static [&'static str] = [
    "sha256",
    "sha384",
    "sha512",
];
```
and use the position in the array as the priority instead.